src/Common/src/System/Runtime/InteropServices/FunctionWrapper.cs

```diff
@@ -74,6 +75,11 @@ internal static partial class FunctionWrapper
                 {
                     return new FunctionLoadResult<T>(FunctionLoadResultKind.Success, Marshal.GetDelegateForFunctionPointer<T>(funcPtr));
                 }
+#else // use managed NativeLibrary API from .NET Core 3 onwards
```

Comment:
What if this code is compiled against `ns2.0`? 

Maybe we should define a constant for `netcoreapp30 or later`

Reply:
Ah, this will break it. I am experimenting with NativeLibrary changes for Windows in a separate branch, it seems like we would need that `netcoreapp30 or later` constant in more than one project. What would be the best place to define such a constant, so it can be shared among multiple projects?

Is there a CI leg for NetStandard 2.0?

Reply:
We currently define these constants in `Directory.Build.props` 

For example, `Corefx.Private.TestUtilities` and `System.Drawing.Common` are built for `netstandard` which as of now means `netstandard2.0` (we have work being done to retarget it to `2.1`. This file is used by `Drawing.Common` however, for `netstandard` I don't see it would be a problem because we just generate a `PlatformNotSupported` assembly, but for `Private.TestUtilities` it could be a problem.

The way a `netstandard` configuration would be chosen for these 2 libraries would be in the `Package All Configurations` leg... or locally if you do, `build -allConfigurations`

Eventhough this code is only part of those 2 assemblies I think it makes sense to protect this code from being used in other configuration where this APIs are not available or used by other library.

Reply:
cc: @ericstj as we've talked about these constant previously.

Reply:
Thanks for the detailed answer @safern!
Since `netcoreapp` TargetGroup and constant correspond to .NET Core 3.0 today and in future the higher version, would it be ok to use `#if !netcoreapp \n dlopen() \n #else \n NativeLibrary.Load() \n #endif`? That is, without defining a new constant. :)

Reply:
But what if, we start building a library intentionally against `netcoreapp3.0`? For example, we add a new API that we want to use in `System.Drawing.Common` we would have to build for `3.0` and for `3.1` so if we add the `netcoreapp3.0` target group explicitly, that would go through the #else statement.

Reply:
> Maybe we should define a constant for netcoreapp30 or later

I prefer to check for the legacy configs that we will likely eventually not care about in conditions like these. It ensures that the old path is not accidentally used for new config, and it makes the legacy configs easy to cleanup once we stop caring about them.

Reply:
That is also true... So if we go that route, then we should use the `!netcoreapp` condition?

Reply:
> So if we go that route, then we should use the !netcoreapp condition?

Reply:
> netstandard I don't see it would be a problem because we just generate a PlatformNotSupported assembly, but for Private.TestUtilities it could be a problem.

The build is green so it does not seem to be a problem for the change as is.

Reply:
Yes I agree. I was just trying to anticipate possible future errors. I'm ok with leaving as is, but I wanted to be consistent, maybe we should add || `netstandard20` to the if def.

Reply:
Seems like we only have `netstandard` TargetGroup: https://github.com/dotnet/corefx/blob/1e3385eb676c249a8358b6b7f86de9d21ad35b4d/Directory.Build.props#L437

Reply:
Yes, that is because we don't have a netstandard2.0 target group yet.